presto-native-execution/src/test/java/com/facebook/presto/nativeworker/TestPrestoNativeAsyncDataCacheCleanupAPI.java

```diff
@@ -0,0 +1,197 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.nativeworker;
+
+import com.facebook.presto.metadata.InternalNode;
+import com.facebook.presto.testing.ExpectedQueryRunner;
+import com.facebook.presto.testing.QueryRunner;
+import com.facebook.presto.tests.AbstractTestQueryFramework;
+import com.facebook.presto.tests.DistributedQueryRunner;
+import org.testng.annotations.Test;
+
+import java.util.Map;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+import static com.facebook.presto.nativeworker.NativeApiEndpointUtils.fetchScalarLongMetrics;
+import static com.facebook.presto.nativeworker.NativeApiEndpointUtils.sendWorkerRequest;
+import static com.facebook.presto.nativeworker.NativeQueryRunnerUtils.createCustomer;
+import static java.lang.Thread.sleep;
+import static org.testng.Assert.assertEquals;
+import static org.testng.Assert.assertNotEquals;
+
+public class TestPrestoNativeAsyncDataCacheCleanupAPI
+        extends AbstractTestQueryFramework
+{
+    private static final String memoryCacheCleanupEndPoint = "/v1/operation/server/clearCache?type=memory";
+    private static final String ssdCacheCleanupEndPoint = "/v1/operation/server/clearCache?type=ssd";
+    private static final String metricsEndPoint = "/v1/info/metrics";
+    private static final String writeToSSDEndPoint = "/v1/operation/server/writeSsd";
+
+    @Override
+    protected QueryRunner createQueryRunner() throws Exception
+    {
+        return PrestoNativeQueryRunnerUtils.createQueryRunner(true, false, true, true);
+    }
+
+    @Override
+    protected ExpectedQueryRunner createExpectedQueryRunner() throws Exception
+    {
+        return PrestoNativeQueryRunnerUtils.createJavaQueryRunner();
+    }
+
+    @Override
+    protected void createTables()
+    {
+        QueryRunner queryRunner = (QueryRunner) getExpectedQueryRunner();
+        createCustomer(queryRunner);
+    }
+
+    @Test
+    public void testAsyncDataCacheCleanup() throws Exception
+    {
+        QueryRunner queryRunner = getQueryRunner();
+        DistributedQueryRunner distributedQueryRunner = (DistributedQueryRunner) queryRunner;
+        Set<InternalNode> workerNodes = getWorkerNodes(distributedQueryRunner);
+
+        // 1. Collect initial cache metrics
+        Metrics initialMetrics = collectCacheMetrics(workerNodes, metricsEndPoint);
+        assertEquals(0, initialMetrics.memoryCacheHits, "Cache hits should be zero initially.");
+        assertEquals(0, initialMetrics.memoryCacheEntries, "Cache entries should be zero initially.");
+
+        // 2. Execute queries to populate cache
+        for (int i = 0; i < 4; i++) {
+            assertQuery("SELECT count(*) FROM customer");
+        }
+        sleep(60000); // Sleep to allow cache updates
```

Comment:
is this 1 minute? would it be a bit too long?

Reply:
The ops are usually very fast only takes 1s or so. Have u ever observed flakiness?